plugins/in_dbus/in_dbus.c

```diff
@@ -0,0 +1,527 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019      The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_input.h>
+#include <fluent-bit/flb_config.h>
+#include <fluent-bit/flb_error.h>
+#include <fluent-bit/flb_utils.h>
+#include <fluent-bit/flb_pack.h>
+#include <fluent-bit/flb_parser.h>
+#include <msgpack.h>
+#include <dbus/dbus.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "in_dbus.h"
+
+struct in_dbus_event {
+    struct mk_event event;
+    struct mk_event_loop* evl;
+    DBusConnection* conn;
+    DBusWatch* watch;
+};
+
+int in_dbus_event_handler(void* data) {
+    struct in_dbus_event* event = data;
+    const unsigned flags = dbus_watch_get_flags(event->watch);
+    if (!dbus_watch_handle(event->watch, flags)) {
+        flb_error("[in_dbus] dbus_watch_handle failed");
```

Comment:
This `[in_dbus]` is repeated a lot -- maybe define wrapper macros?
```
#define error(...) flb_error("[in_dbus] " __VA_ARGS__)
```

Reply:
Looks like they [permit using the `##` C extension](https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_log.h#L89), so this sort of thing is probably stricter/better: 
```
#define error(fmt, ...) flb_error("[in_dbus] " fmt, ##__VA_ARGS__)
```